com.unity.mobile.notifications/Runtime/iOS/Plugins/UnityNotificationManager.m

```diff
@@ -179,8 +184,8 @@ - (void)userNotificationCenter:(UNUserNotificationCenter *)center willPresentNot
         showInForeground = [[notification.request.content.userInfo objectForKey: @"showInForeground"] boolValue];
     }
 
-    freeiOSNotificationData(notificationData);
-    notificationData = NULL;
+    if (haveNotificationData)
+        freeiOSNotificationData(&notificationData);
```

Comment:
Why do we have to free the memory on the stack?

Reply:
Oh, you want to release the heap memory held by `notificationData`, can we add some comment here?

Reply:
Why we have a separate function freeiOSNotificationData, why not simply have a function inside iOSNotificationData ?

Because this code looks misleading, it looks like it's freeing the whole notificationData while in reality it frees the memory for fields.

Can you move freeiOSNotificationData to iOSNotificationData instead? That would be more clearer

Reply:
That would require substantial changes in the package. This and most others are Object-C files, not Objective-C++, we cannot have methods in structs.

Reply:
oh didn't know that... do you know of any reason why all these files are objc and not objc++?

Reply:
I can only guess, that it was convenient at the time: XCode let's you add new Obj-C file, but not Obj-C++, for the later you have to manually rename the file extension. And you don't need to write extern "C" clutter for functions called from C# in Obj-C. The price is that you can't use C++ features, but there's almost no real need for any.